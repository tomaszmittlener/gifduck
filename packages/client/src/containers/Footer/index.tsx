import React, { FunctionComponent } from 'react'
import styled from 'styled-components'

interface HeaderProps {
  className?: string
  testId?: string
}

const Container = styled.footer`
  margin: auto 0 0 0;
  width: 100%;
  min-height: ${({ theme }) => theme.ms(20)};
  background-color: ${({ theme }) => theme.colors.accent.default};
  align-items: center;
  justify-content: center;
  display: flex;
  padding: ${({ theme }) => theme.ms(3)} 0px;
`

const Text = styled.span`
  color: ${({ theme }) => theme.colors.accent.defaultInverted};
  font-family: ${({ theme }) => theme.typography.fonts.secondary};
`

const Footer: FunctionComponent<HeaderProps> = ({ className, testId }) => {
  return (
    <Container data-testid={testId} className={className}>
      <Text>@gifduck by Tomasz Mittlener</Text>
    </Container>
  )
}

export default Footer
