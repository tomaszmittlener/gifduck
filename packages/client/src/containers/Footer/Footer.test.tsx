import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import Theme from 'styles/theme'

import Footer from './'

const defaultProps = {
  className: 'class-name',
  testId: 'test-id',
}

afterEach(cleanup)

describe('Footer', () => {
  test('renders correctly', async () => {
    const { container } = render(
      <ThemeProvider theme={Theme}>
        <Footer />
      </ThemeProvider>,
    )
    expect(container).toMatchSnapshot()
  })

  test('passes className to container', async () => {
    const { getByTestId } = render(
      <ThemeProvider theme={Theme}>
        <Footer {...defaultProps} />
      </ThemeProvider>,
    )

    expect(getByTestId(defaultProps.testId)).toHaveClass(defaultProps.className)
  })
})
