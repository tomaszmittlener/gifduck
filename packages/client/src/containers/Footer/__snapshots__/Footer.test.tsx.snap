// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
.c0 {
  margin: auto 0 0 0;
  width: 100%;
  min-height: 10.5450938424492rem;
  background-color: #F2C94C;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 1.423828125rem 0px;
}

.c1 {
  color: #0D2E4D;
  font-family: Pacifico,sans-serif;
}

<div>
  <footer
    class="c0"
  >
    <span
      class="c1"
    >
      @gifduck by Tomasz Mittlener
    </span>
  </footer>
</div>
`;
